src/_pytest/_code/code.py

```diff
@@ -976,6 +1035,7 @@ def __init__(self, lines, reprfuncargs, reprlocals, filelocrepr, style):
 
     def toterminal(self, tw) -> None:
         if self.style == "short":
+            assert self.reprfileloc is not None
```

Comment:
Could (additionally) be checked/validated in `__init__`?  But it's fine like that also.

Reply:
Wouldn't happen unless the code is buggy; seems fine to only do it here.